File: Java/src/napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author Oscar
 */
public class CultistPlayer extends Player{
    
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        totalCultistPlayers++;
        myCultistCard = c;
        
    }
    
    @Override
    protected int getCombatLevel(){
        int trunc = (int) (super.getCombatLevel() + 0.7 * super.getCombatLevel() + myCultistCard.getGainedLevels()*totalCultistPlayers);
        return trunc;
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    @Override
    protected Treasure giveMeATreasure(){
        Treasure t;
        int indice_aleatorio = (int) (Math.random() * visibleTreasures.size());
        t = visibleTreasures.get(indice_aleatorio);
        visibleTreasures.remove(indice_aleatorio);
        return t;
    }
    
    @Override
    protected boolean canYouGiveMeATreasure(){
        boolean bool = true;
        
        if(getVisibleTreasures().isEmpty())
            bool = false;
        
        return bool;
    }
    
    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    
    
}

File: Ruby/lib/CultistPlayer.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  
  require_relative "Cultist"
  require_relative "Player.rb"
  
  class CultistPlayer < Player
        @@totalCultistPlayers = 0
    
    def initialize( c, p ) 
       @myCultistCard = c
       @@totalCultistPlayers += 1
       self.copy(p)
    end
    
    def self.getTotalCultistPlayers
      @@totalCultistPlayers
    end
    
    protected
    def getCombatLevel()
      return (super + 0.7 * super + @myCulistCard.getLevelsGained*@@totalCultistPlayers)
        #return (Player.instance_method(:getCombatLevel).bind(self).call*1.7+@myCultistCard.gainedLevels*@@totalCultistPlayers).to_i
    end
    
    protected
    def getOponentLevel(m)
      m.getCombatLevelAgainstCultistPlayer()
    end
    
    protected
    def shouldConvert()
      false
    end
    
    protected
    def giveMeaTreasure()
      @visibleTreasures.delete_at(rand(@visibleTreasures.size))
    end
    
    protected
    def canYouGiveMeATreasure()
      !@visibleTreasures.empty?
    end
    
    
  end

end